analysis.py

```diff
@@ -110,9 +110,23 @@ def gpggaParse(self, gpgga, line):
         gpgga[6] = line[8]
         gpgga[7] = line[9] + line[10]
         gpgga[8] = line[11] + line[12]
-        gpgga[9] = line[13]
-        print(gpgga)
-        return gpgga
+        gpgga[9] = line[14]
+        
+        if self.verifyChecksum(line, gpgga[9]):
+            return gpgga
+        else:
+            raise Exception("Checksum is incorrect! \n" +
```

Comment:
Done -- moved the exception raising to the verifyChecksum function. Code is cleaner this way anyways.